external_search.py

```diff
@@ -33,114 +42,176 @@ def __init__(self, url=None, works_index=None):
                 if not works_index:
                     works_index = integration.get(
                         Configuration.ELASTICSEARCH_INDEX_KEY
-                    ) or None
+                    )
 
                 if not url:
                     if not integration:
                         return
                     url = integration[Configuration.URL]
 
-            use_ssl = url and url.startswith('https://')
+            if not url:
+                raise Exception("Cannot connect to Elasticsearch cluster.")
+            use_ssl = url.startswith('https://')
             self.log.info(
                 "Connecting to index %s in Elasticsearch cluster at %s", 
                 works_index, url
             )
             ExternalSearchIndex.__client = Elasticsearch(
                 url, use_ssl=use_ssl, timeout=20, maxsize=25
             )
-            ExternalSearchIndex.__client.works_index = works_index
-            if not url:
-                raise Exception("Cannot connect to Elasticsearch cluster.")
 
-        self.works_index = self.__client.works_index
         self.indices = self.__client.indices
         self.search = self.__client.search
         self.index = self.__client.index
         self.delete = self.__client.delete
         self.exists = self.__client.exists
+
+        # Sets self.works_index and self.works_alias values.
+        # Document upload runs against the works_index.
+        # Search queries run against works_alias.
+        self.works_index = None
+        self.works_alias = None
+        self.set_works_index_and_alias(works_index)
+
         def bulk(docs, **kwargs):
             return elasticsearch_bulk(self.__client, docs, **kwargs)
         self.bulk = bulk
-            
+
+    def set_works_index_and_alias(self, current_alias):
+        """Finds or creates the works_index and works_alias based on
+        provided configuration.
+        """
+        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
+        found = not (index_details.get('status')==404 or 'error' in index_details)
+
+        def _set_works_index(name):
+            self.works_index = self.__client.works_index = name
+
+        if found:
+            # We found an index for the alias in configuration. Assume
+            # there is only one.
+            _set_works_index(index_details.keys()[0])
+        else:
+            if current_alias.endswith(self.CURRENT_ALIAS_SUFFIX):
+                # The alias culled from configuration is intended to be
+                # a current alias, but we couldn't find an index with that
+                # alias.
+                raise ValueError(
+                    "External search alias '%s' not found." % current_alias)
```

Comment:
I think that's fine, now that we're not moving the alias if it's already pointing to something else.